DenonAVR Config Flow

starkillerOG opened the pull request:
!!!WARNING!!!
This can not be merged untill the denonavr library has been pushed to a new version.
This PR first needs to be merged: https://github.com/scarface-4711/denonavr/pull/132

<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Breaking change
<!--
  If your PR contains a breaking change for existing users, it is important
  to tell them what breaks, how to make it work again and why we did this.
  This piece of text is published with the release notes, so it helps if you
  write it towards our users, not us.
  Note: Remove this section if this PR is NOT a breaking change.
-->
YAML configuration of the denonavr component has been depricated, please remove the denonavr platform from your configuration.yaml file.
The denonavr component now uses the new Config Flow.
Denon and Marantz receivers schould be discovered automatically and show up under "Configuration"-> "Integrations", from there hit configure and go through the steps, all previous setting options are still available during this setup.
If your receiver does not show up automatically, go to "Configuration"-> "Integrations"-> press the "+" icon -> search for "denonavr" and enter the setup.

## Proposed change
<!-- 
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
- Implement Config Flow
- Implement the new SSDP discovery using Config Flow
- Set device_ids and set unique ids for the receiver and there zones.
- Remove the depricated discovery implementation.

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box! 
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [x] New feature (which adds functionality to an existing integration)
- [x] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Example entry for `configuration.yaml`:
<!--
  Supplying a configuration snippet, makes it easier for a maintainer to test
  your PR. Furthermore, for new integrations, it gives an impression of how
  the configuration would look like.
  Note: Remove this section if this PR does not have an example entry.
-->

YAML depricated --> Config Flow :)

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes https://github.com/home-assistant/core/issues/35189 and https://github.com/home-assistant/core/issues/34871
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [x] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [x] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [ ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

starkillerOG committed:
Merge pull request #1 from home-assistant/dev

Update my own fork

starkillerOG committed:
DenonAVR Config Flow

starkillerOG committed:
DenonAVR Config Flow

starkillerOG committed:
DenonAVR Config Flow

starkillerOG committed:
DenonAVR Config Flow

starkillerOG committed:
DenonAVR Config Flow

starkillerOG committed:
DenonAVR Config Flow

starkillerOG committed:
DenonAVR Config Flow

starkillerOG committed:
DenonAVR Config Flow

starkillerOG committed:
implement DenonAVR Config Flow

starkillerOG committed:
implement DenonAVR Config Flow

starkillerOG committed:
DenonAvr Config Flow

starkillerOG committed:
DenonAvr Config Flow

starkillerOG committed:
fix json

starkillerOG committed:
fix json

starkillerOG committed:
fix config flow

starkillerOG committed:
fix config flow

starkillerOG committed:
DenonAvr add unique id & device id

starkillerOG committed:
DenonAvr add unique id & device id

starkillerOG committed:
DenonAVR Config Flow

starkillerOG committed:
DenonAVR Config Flow

starkillerOG committed:
Finish DenonAVR Config Flow

starkillerOG committed:
Finish DenonAVR Config Flow

starkillerOG committed:
DenonAVR Config Flow, Add SSDP discovery

starkillerOG committed:
DenonAVR Config Flow, Add SSDP discovery

starkillerOG commented:
Still need to implement the tests for the config flow

starkillerOG committed:
Update docstring

starkillerOG committed:
Update docstring

starkillerOG committed:
fix spelling mistakes

starkillerOG committed:
fix spelling mistakes

OnFreund committed:
Use suggestd_value instead of default in Monoprice options flow (#35107)

starkillerOG committed:
attempt to fix test

starkillerOG committed:
attempt to fix test

starkillerOG commented:
@MartinHjelmare I can not seem to get the test_media_player.py to work again.
Could you have a look at what I am doing wrong?

If I run the test I get:
```

――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――― test_get_command[pyloop] ――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――

hass = <homeassistant.core.HomeAssistant object at 0x7fbb271dbed0>

    async def test_get_command(hass):
        """Test generic command functionality."""
        mock_receiver = get_mock_receiver()
        await setup_denonavr(hass, mock_receiver)

        data = {
            ATTR_ENTITY_ID: ENTITY_ID,
            ATTR_COMMAND: "test",
        }
        await hass.services.async_call(DOMAIN, SERVICE_GET_COMMAND, data)
        await hass.async_block_till_done()

>       mock_receiver.send_get_command.assert_called_with("test")

tests/components/denonavr/test_media_player.py:99:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

_mock_self = <Mock name='mock.send_get_command' id='140441678800848'>, args = ('test',), kwargs = {}, expected = "send_get_command('test')"

    def assert_called_with(_mock_self, *args, **kwargs):
        """assert that the mock was called with the specified arguments.

        Raises an AssertionError if the args and keyword args passed in are
        different to the last call to the mock."""
        self = _mock_self
        if self.call_args is None:
            expected = self._format_mock_call_signature(args, kwargs)
>           raise AssertionError('Expected call: %s\nNot called' % (expected,))
E           AssertionError: Expected call: send_get_command('test')
E           Not called

/usr/lib/python3.7/unittest/mock.py:864: AssertionError
------------------------------------------------------------------- Captured stderr setup -------------------------------------------------------------------
DEBUG:asyncio:Using selector: EpollSelector
-------------------------------------------------------------------- Captured log setup ---------------------------------------------------------------------
DEBUG    asyncio:selector_events.py:53 Using selector: EpollSelector
------------------------------------------------------------------- Captured stderr call --------------------------------------------------------------------
INFO:homeassistant.loader:Loaded test from custom_components.test
INFO:homeassistant.loader:Loaded test_package from custom_components.test_package
INFO:homeassistant.loader:Loaded denonavr from homeassistant.components.denonavr
INFO:homeassistant.setup:Setting up denonavr
DEBUG:homeassistant.core:Bus:Handling <Event service_registered[L]: domain=denonavr, service=get_command>
INFO:homeassistant.setup:Setup of domain denonavr took 0.0 seconds.
DEBUG:homeassistant.core:Bus:Handling <Event component_loaded[L]: component=denonavr>
INFO:homeassistant.loader:Loaded media_player from homeassistant.components.media_player
INFO:homeassistant.loader:Loaded http from homeassistant.components.http
INFO:homeassistant.setup:Setting up http
INFO:homeassistant.setup:Setup of domain http took 0.0 seconds.
DEBUG:homeassistant.core:Bus:Handling <Event component_loaded[L]: component=http>
INFO:homeassistant.setup:Setting up media_player
DEBUG:homeassistant.core:Bus:Handling <Event service_registered[L]: domain=media_player, service=turn_on>
DEBUG:homeassistant.core:Bus:Handling <Event service_registered[L]: domain=media_player, service=turn_off>
DEBUG:homeassistant.core:Bus:Handling <Event service_registered[L]: domain=media_player, service=toggle>
DEBUG:homeassistant.core:Bus:Handling <Event service_registered[L]: domain=media_player, service=volume_up>
DEBUG:homeassistant.core:Bus:Handling <Event service_registered[L]: domain=media_player, service=volume_down>
DEBUG:homeassistant.core:Bus:Handling <Event service_registered[L]: domain=media_player, service=media_play_pause>
DEBUG:homeassistant.core:Bus:Handling <Event service_registered[L]: domain=media_player, service=media_play>
DEBUG:homeassistant.core:Bus:Handling <Event service_registered[L]: domain=media_player, service=media_pause>
DEBUG:homeassistant.core:Bus:Handling <Event service_registered[L]: domain=media_player, service=media_stop>
DEBUG:homeassistant.core:Bus:Handling <Event service_registered[L]: domain=media_player, service=media_next_track>
DEBUG:homeassistant.core:Bus:Handling <Event service_registered[L]: domain=media_player, service=media_previous_track>
DEBUG:homeassistant.core:Bus:Handling <Event service_registered[L]: domain=media_player, service=clear_playlist>
DEBUG:homeassistant.core:Bus:Handling <Event service_registered[L]: domain=media_player, service=volume_set>
DEBUG:homeassistant.core:Bus:Handling <Event service_registered[L]: domain=media_player, service=volume_mute>
DEBUG:homeassistant.core:Bus:Handling <Event service_registered[L]: domain=media_player, service=media_seek>
DEBUG:homeassistant.core:Bus:Handling <Event service_registered[L]: domain=media_player, service=select_source>
DEBUG:homeassistant.core:Bus:Handling <Event service_registered[L]: domain=media_player, service=select_sound_mode>
DEBUG:homeassistant.core:Bus:Handling <Event service_registered[L]: domain=media_player, service=play_media>
DEBUG:homeassistant.core:Bus:Handling <Event service_registered[L]: domain=media_player, service=shuffle_set>
INFO:homeassistant.setup:Setup of domain media_player took 0.0 seconds.
DEBUG:homeassistant.core:Bus:Handling <Event component_loaded[L]: component=media_player>
INFO:homeassistant.components.media_player:Setting up media_player.denonavr
INFO:homeassistant.components.denonavr.media_player:<Mock name='mock.manufacturer' id='140441678697744'> receiver at host <Mock name='mock.host' id='140441678697936'> initialized
DEBUG:homeassistant.core:Bus:Handling <Event device_registry_updated[L]: action=create, device_id=b364065515e14a8dbbc17d124759aeae>
INFO:homeassistant.helpers.entity_registry:Registered new media_player.denonavr entity: media_player.fake
DEBUG:homeassistant.core:Bus:Handling <Event entity_registry_updated[L]: action=create, entity_id=media_player.fake>
DEBUG:homeassistant.core:Bus:Handling <Event state_changed[L]: entity_id=media_player.fake, old_state=None, new_state=<state media_player.fake=<Mock name='mock.state' id='140441678655248'>; volume_level=1.0, is_volume_muted=<Mock name='mock.muted' id='140441678655056'>, media_content_type=channel, media_title=<Mock name='mock.input_func' id='140441678655120'>, media_artist=<Mock name='mock.artist' id='140441678696720'>, media_album_name=<Mock name='mock.album' id='140441678696912'>, source=<Mock name='mock.input_func' id='140441678655120'>, sound_mode=<Mock name='mock.sound_mode' id='140441678697488'>, friendly_name=fake, supported_features=69004 @ 2020-05-03T15:54:00.549888+00:00>>
DEBUG:homeassistant.core:Bus:Handling <Event call_service[L]: domain=denonavr, service=get_command, service_data=entity_id=media_player.fake, command=test>
--------------------------------------------------------------------- Captured log call ---------------------------------------------------------------------
INFO     homeassistant.loader:loader.py:209 Loaded test from custom_components.test
INFO     homeassistant.loader:loader.py:209 Loaded test_package from custom_components.test_package
INFO     homeassistant.loader:loader.py:209 Loaded denonavr from homeassistant.components.denonavr
INFO     homeassistant.setup:setup.py:154 Setting up denonavr
DEBUG    homeassistant.core:core.py:593 Bus:Handling <Event service_registered[L]: domain=denonavr, service=get_command>
INFO     homeassistant.setup:setup.py:188 Setup of domain denonavr took 0.0 seconds.
DEBUG    homeassistant.core:core.py:593 Bus:Handling <Event component_loaded[L]: component=denonavr>
INFO     homeassistant.loader:loader.py:209 Loaded media_player from homeassistant.components.media_player
INFO     homeassistant.loader:loader.py:209 Loaded http from homeassistant.components.http
INFO     homeassistant.setup:setup.py:154 Setting up http
INFO     homeassistant.setup:setup.py:188 Setup of domain http took 0.0 seconds.
DEBUG    homeassistant.core:core.py:593 Bus:Handling <Event component_loaded[L]: component=http>
INFO     homeassistant.setup:setup.py:154 Setting up media_player
DEBUG    homeassistant.core:core.py:593 Bus:Handling <Event service_registered[L]: domain=media_player, service=turn_on>
DEBUG    homeassistant.core:core.py:593 Bus:Handling <Event service_registered[L]: domain=media_player, service=turn_off>
DEBUG    homeassistant.core:core.py:593 Bus:Handling <Event service_registered[L]: domain=media_player, service=toggle>
DEBUG    homeassistant.core:core.py:593 Bus:Handling <Event service_registered[L]: domain=media_player, service=volume_up>
DEBUG    homeassistant.core:core.py:593 Bus:Handling <Event service_registered[L]: domain=media_player, service=volume_down>
DEBUG    homeassistant.core:core.py:593 Bus:Handling <Event service_registered[L]: domain=media_player, service=media_play_pause>
DEBUG    homeassistant.core:core.py:593 Bus:Handling <Event service_registered[L]: domain=media_player, service=media_play>
DEBUG    homeassistant.core:core.py:593 Bus:Handling <Event service_registered[L]: domain=media_player, service=media_pause>
DEBUG    homeassistant.core:core.py:593 Bus:Handling <Event service_registered[L]: domain=media_player, service=media_stop>
DEBUG    homeassistant.core:core.py:593 Bus:Handling <Event service_registered[L]: domain=media_player, service=media_next_track>
DEBUG    homeassistant.core:core.py:593 Bus:Handling <Event service_registered[L]: domain=media_player, service=media_previous_track>
DEBUG    homeassistant.core:core.py:593 Bus:Handling <Event service_registered[L]: domain=media_player, service=clear_playlist>
DEBUG    homeassistant.core:core.py:593 Bus:Handling <Event service_registered[L]: domain=media_player, service=volume_set>
DEBUG    homeassistant.core:core.py:593 Bus:Handling <Event service_registered[L]: domain=media_player, service=volume_mute>
DEBUG    homeassistant.core:core.py:593 Bus:Handling <Event service_registered[L]: domain=media_player, service=media_seek>
DEBUG    homeassistant.core:core.py:593 Bus:Handling <Event service_registered[L]: domain=media_player, service=select_source>
DEBUG    homeassistant.core:core.py:593 Bus:Handling <Event service_registered[L]: domain=media_player, service=select_sound_mode>
DEBUG    homeassistant.core:core.py:593 Bus:Handling <Event service_registered[L]: domain=media_player, service=play_media>
DEBUG    homeassistant.core:core.py:593 Bus:Handling <Event service_registered[L]: domain=media_player, service=shuffle_set>
INFO     homeassistant.setup:setup.py:188 Setup of domain media_player took 0.0 seconds.
DEBUG    homeassistant.core:core.py:593 Bus:Handling <Event component_loaded[L]: component=media_player>
INFO     homeassistant.components.media_player:entity_platform.py:165 Setting up media_player.denonavr
INFO     homeassistant.components.denonavr.media_player:media_player.py:67 <Mock name='mock.manufacturer' id='140441678697744'> receiver at host <Mock name='mock.host' id='140441678697936'> initialized
DEBUG    homeassistant.core:core.py:593 Bus:Handling <Event device_registry_updated[L]: action=create, device_id=b364065515e14a8dbbc17d124759aeae>
INFO     homeassistant.helpers.entity_registry:entity_registry.py:276 Registered new media_player.denonavr entity: media_player.fake
DEBUG    homeassistant.core:core.py:593 Bus:Handling <Event entity_registry_updated[L]: action=create, entity_id=media_player.fake>
DEBUG    homeassistant.core:core.py:593 Bus:Handling <Event state_changed[L]: entity_id=media_player.fake, old_state=None, new_state=<state media_player.fake=<Mock name='mock.state' id='140441678655248'>; volume_level=1.0, is_volume_muted=<Mock name='mock.muted' id='140441678655056'>, media_content_type=channel, media_title=<Mock name='mock.input_func' id='140441678655120'>, media_artist=<Mock name='mock.artist' id='140441678696720'>, media_album_name=<Mock name='mock.album' id='140441678696912'>, source=<Mock name='mock.input_func' id='140441678655120'>, sound_mode=<Mock name='mock.sound_mode' id='140441678697488'>, friendly_name=fake, supported_features=69004 @ 2020-05-03T15:54:00.549888+00:00>>
DEBUG    homeassistant.core:core.py:593 Bus:Handling <Event call_service[L]: domain=denonavr, service=get_command, service_data=entity_id=media_player.fake, command=test>

 tests/components/denonavr/test_media_player.py ⨯                                                                                             100% ██████████
================================================================== short test summary info ==================================================================
FAILED tests/components/denonavr/test_media_player.py::test_get_command[pyloop] - AssertionError: Expected call: send_get_command('test')

Results (0.24s):
       1 failed
         - tests/components/denonavr/test_media_player.py:87 test_get_command[pyloop]

```

MartinHjelmare commented on tests/components/denonavr/test_media_player.py:
```diff
@@ -1,51 +1,89 @@
+    
+    assert await async_setup_component(hass, DOMAIN, {}) is True
+
+    hass.data[DOMAIN] = {config_entry.entry_id: mock_receiver}
```
Don't mock integration code in the tests. Patch the library as was done before.

JPHutchins replied:
@starkillerOG I will pull your new version of the Denon library when I have a chance and then follow @MartinHjelmare 's advice above.  Hopefully with HA using your new denon PR we will get more clarity.  I may not have time until Thursday, Friday or Saturday.

starkillerOG replied:
@JPHutchins while I was working on the Config Flow test I figured out what was wrong with this test and I have learned what fixtures are:)
So now I have already fixed the test.

Thank you anyway for offering your help @JPHutchins

LordMike committed:
Simplify MQTT light brightness logic (#35097)

balloob committed:
Use asynctest-mock in most places (#35109)

* Use asynctest-mock in most places

* Fix broken patch in pilight

bdraco committed:
Make alexa and google aware of DEVICE_CLASS_GATE (#35103)

balloob committed:
Add type to device registry (#35095)

MartinHjelmare committed:
Add required features to cover service registration (#35073)

* Fix typing in component service method

* Add required features to cover service registration

* Fix template cover tilt features

* Delint template cover tests

frenck committed:
Upgrade spotipy to 2.12.0 (#35149)

frenck committed:
Upgrade numpy to 1.18.4 (#35150)

ronaldheft committed:
Add support for refreshing synology_dsm sensors (#35141)

* Add support for refreshing synology_dsm sensors

Now supports `home_assistant.update_entity` service

* Don't immediately update sensors on add

This fixes all sensors being forcefully refreshed when setup.

balloob committed:
Hue: Guard for when there is no brightness (#35151)

Quentame committed:
Bump pyiCloud to 0.9.7 + do not warn when pending devices (#35156)

Tho85 committed:
Return emulated hue id for update requests (#35139)

* Return emulated hue id for update requests

* Emulated Hue: Rename function argument to match its content

* Use HTTP status consts throughout the test

* Use HTTP status consts in UPnP test

mtdcr committed:
Support multiple EDL21 meters (#33594)

* edl21: Add 'name' option

* edl21: Include domain and electricity ID in unique ID

As a side-effect, this makes electricity IDs mandatory, i.e. devices
which don't transmit their ID won't register any sensors.

* edl21: Implement upgrade path for unique IDs

* Update homeassistant/components/edl21/sensor.py

Drop DOMAIN from unique ID.

Co-Authored-By: J. Nick Koston <nick@koston.org>

Co-authored-by: J. Nick Koston <nick@koston.org>

Tho85 committed:
Improve emulated_hue compatibility with newer systems (#35148)

* Make emulated hue detectable by Busch-Jaeger free@home SysAP

* Emulated hue: Remove unnecessary host line from UPnP response

* Test that external IPs are blocked for config route

* Add another test for unauthorized users

* Change hue username to nouser

nouser seems to be used by the official Hue Bridge v1 Android app and is
used by other projects as well

Co-authored-by: J. Nick Koston <nick@koston.org>

frenck committed:
Upgrade pyupgrade to v2.3.0 (#35159)

frenck committed:
Fix Canary KeyError: 'ffmpeg_arguments' (#35158)

BKPepe committed:
Upgrade cryptography to 2.9.2 (#35084)

davidjb committed:
scrape: extract strings from new non-text tags (#35021)

With the upgrade to beautifulsoup4 to 4.9.0 (#34007), certain tags
(`<style>`, `<script>` and `<template>`) are no longer treated as having
text content (see
https://www.crummy.com/software/BeautifulSoup/bs4/doc/#comments-and-other-special-strings
and reported bug https://bugs.launchpad.net/beautifulsoup/+bug/1868861)
meaning the content of these types of tags became inaccessible to HA.

Where the previous code could access `.text` on the tag, bs4 4.9 now
yields an empty string; these types of tags require accesing `.string`
instead.  This PR checks the tag name (which will aalways be lowercase
given how the parser works;
https://www.crummy.com/software/BeautifulSoup/bs4/doc/#other-parser-problems)
and applies this different access strategy to get the content of the
HTML tag.  All other tags are handled in the original manner.

fabaff committed:
Upgrade zeroconf to 0.26.0 (#35176)

fabaff committed:
Upgrade alpha_vantage to 2.2.0 (#35172)

shenxn committed:
Remove `certificate` configuration variable from roomba (#35162)

* Remove certificate option from roomba

* Fix roomba test

* Add back certificate with deprecated flag

* Remove invalidation_version

fabaff committed:
Upgrade pytz to 2020.1 (#35174)

rajlaud committed:
Squeezebox add query and sync (#31748)

* Add query and sync

* Update description of call_query

* Remove backup files accidentally committed to repo

* Update after pysqueezebox refactor

* Use entity service helper

* Implement suggested changes

* Fix linter error in services.yaml

* Fix long lines in services.yaml

Kane610 committed:
UniFi - Improve signalling and handling of adding new entities (#34925)

* Change signalling and handling of adding devices

* Bump aiounifi to v19
Use the proper variable for roamradio

StevenLooman committed:
Simplification of upnp component (#35191)

amelchio committed:
Some Sonos fixes (#35115)

balloob committed:
Extract instance ID helper from updater (#35043)

dmulcahey committed:
Group by endpoints and not by devices for ZHA Zigbee groups (#34583)

* start implementation
* handle members correctly
* fix group member info
* align groupable devices with group members
* handle group endpoint adding and removing
* update add group
* update group and group member
* update create group
* remove domain check
* update test
* remove temporary 2nd groupable device api
* update test
* rename validator - review comment
* fix test that was never running
* additional testing
* fix coordinator descriptors
* remove check that was done twice
* update test
* Use AsyncMock()
Co-authored-by: Alexei Chetroi <lexoid@gmail.com>

pnbruckner committed:
Move life360 I/O out of event loop in config flow (#35193)

bachya committed:
Allow use of common entity update service in AirVisual (#35203)

Quentame committed:
Fix Synology DSM sensor to be False or 0 (#35208)

emontnemery committed:
Make sure MQTT light brightness is not rounded to 0 (#35207)

xdissent committed:
Add homekit camera support (#32527)

* Add homekit camera support

* Cleanup pyhapcamera inheritance

* Add camera to homekit manifest

* Use upstream pyhap server handler in homekit

* Remove unused homekit constants

* Fix lint errors in homekit camera

* Update homekit camera log messages

* Black after conflict fixes

* More conflict fixes

* missing srtp

* Allow streaming retry when ffmpeg fails to connect

* Fix inherit of camera config, force kill ffmpeg on failure

* Fix audio (Home Assistant only comes with OPUS)

* Fix audio (Home Assistant only comes with OPUS)

* Add camera to the list of supported domains.

* add a test for camera creation

* Add a basic test (still needs more as its only at 44% cover)

* let super handle reconfigure_stream

* Remove scaling as it does not appear to be needed and causes artifacts

* Some more basic tests

* make sure no exceptions when finding the source from the entity

* make sure the bridge forwards get_snapshot

* restore full coverage to accessories.py

* revert usage of super for start/stop stream

* one more test

* more mocking

* Remove -tune zerolatency, disable reconfigure_stream

* Restore -tune zerolatency

Co-authored-by: John Carr <john.carr@unrouted.co.uk>
Co-authored-by: J. Nick Koston <nick@koston.org>

shbatm committed:
ISY994 migration to PyISY v2 (Structure Changes to enable upgrade, Part 1) (#35212)

* ISY994 Structure updates in prep for PyISYv2 (Part 1)

- Correct node categorization.
- Move constants to separate file.
- Consolidate Logging to Constants file.
- Use Home Assistant Constants where possible.
- Use string literals where possible.
- Rename "domain" to "platform" in most places.
- Add additional device support (NODE_FILTER updates).

* Update categorize_programs per review

* add @shbatm as codeowner for ISY994 integration

balloob committed:
Fix AdGuard device info (#35217)

dgomes committed:
Fix utility_meter calibration with float values (#35186)

escoand committed:
Catch samsungtv timeout exception (#35205)

frenck committed:
Fix UVC doing I/O inside the event loop (#35169)

cgarwood committed:
Add zwave_mqtt sensor platform (#35135)

* Add sensor platform

* Fix signal name

* Add sensor discovery schema

* Add test for disabled entities

* Add test for enabling advanced sensor

* Add additional fake sensors and tests for device classes

* More device class tests

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

Adminiuga committed:
Bump up pyserial dependency (#35201)

balloob committed:
Add config flow base strings (#34523)

ziv1234 committed:
Strict creation of the config for dynalite (#34663)

shbatm committed:
ISY994 Structure updates in prep for PyISYv2 (Part 2) (#35225)

- Move core Entity Class definitions to separate file.
- New ISYNodeEntity and ISYProgramEntity base classes.
- Update inheriances on other classes.
- Move helper functions to separate file.

balloob committed:
Fix string

DavidMStraub committed:
Add Home Connect integration (#29214)

gadgetmobile committed:
Add BleBox integration (#32664)

Co-Authored-By: J. Nick Koston <nick@koston.org>

starkillerOG committed:
Add DenonAVR Config Flow tests

starkillerOG committed:
Add DenonAVR Config Flow tests

starkillerOG committed:
black + use fixture to reduce duplication

starkillerOG committed:
black + use fixture to reduce duplication

bieniu commented on homeassistant/components/denonavr/__init__.py:
```diff
@@ -33,3 +35,40 @@ def service_handler(service):
+        entry.data[CONF_ZONE3],
+    )
+    if not await connect_denonavr.async_connect_receiver():
+        return False
```
I think it's better to use `raise ConfigEntryNotReady` than `return False`.

starkillerOG replied:
Thanks for the tip

starkillerOG committed:
fix test_media_player

starkillerOG committed:
fix test_media_player

starkillerOG committed:
raise ConfigEntryNotReady on error

starkillerOG committed:
raise ConfigEntryNotReady on error

starkillerOG committed:
rebase + import patch using test.async_mock

starkillerOG committed:
Merge branch 'patch-23' of https://github.com/starkillerOG/home-assistant into patch-23

starkillerOG committed:
Merge branch 'dev' into patch-23

starkillerOG commented:
Sorry I did something wrong trying to rebase this branch, it is now broken....
Therefore I close this PR and have created a new PR with a fresh branch here: https://github.com/home-assistant/core/pull/35255
